### PR TITLE
Add command to update vs code settings

### LINKED ISSUE #13: add command to update workspace/global configuration
This would setup vscode to recommend values -- indent style, ruby linting, editor behavior, c/cpp globals...

### PR DESCRIPTION
**Pull request by g-arjones**
Closes #13

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -147,4 +150,39 @@ export class ConfigManager
                 JSON.stringify(data, null, 4), this._defaultWriteOptions);
         }
     }
+    suggestedSettings(): any
+    {
+        return {
+            "C_Cpp.errorSquiggles": "Enabled",
```
Comment (doudou) [inline src/config.ts]: I don't believe the cosmetic changes have their place here. The whitespace stuff, intellisense and ruby lint, makes a lot of sense, but that and the debug config are really personal preference.

Comment (g-arjones) [reply src/config.ts]: OK.. Anything other than the error squiggles and debug.* ?

Comment (doudou) [reply src/config.ts]: I don't see anything else. The rest seem to be either functional settings, or cosmetic changes that are specific to Rock (the 4 space indent thing)

Comment (g-arjones) [reply src/config.ts]: Updated.

**Comment by doudou** [comment]
Do you think it would make sense to split between global and workspace-local settings ? This way, people would be able to use Rock settings in Rock workspaces, and leave their global config alone.

**Comment by g-arjones** [comment]
Not sure about "splitting" (they all may be applied globally or to the workspace). I think it makes sense to apply to the workspace and I was going to do that but I didn't like vscode's behavior when the user is working on workspace that is not saved: it silently does nothing.

We could prompt the user whether it should be applied globally or to the workspace... Thoughts?

**Comment by doudou** [comment]
> We could prompt the user whether it should be applied globally or to the workspace... Thoughts?

Either that, or provide two commands. The only possibly annoying thing is that some configuration options seem to be unavailable in narrower scopes (at least, we can't change them from the editor).

**Comment by g-arjones** [comment]
Just checked and these are all fine either globally or on workspace scope. I will update to add a picker so the user chooses where he/she wants it..

**Comment by doudou** [comment]
> I didn't like vscode's behavior when the user is working on workspace that is not saved: it silently does nothing.

I've done some tests, the behavior I see is a bit more nuanced:
- when the workspace is empty, it silently does nothing (which kind of makes sense since there is no workspace). This can be detected since `vscode.workspace.name` is undefined in this case.
- when the workspace is not empty yet not saved, the settings are updated nonetheless.
Yeah, noticed that as well. Moreover, I don't see any way in the API to find out whether the workspace has been saved or not. The behavior is strange-ish: if there is only one folder, it saves within this folder. If there is more than one folder, it saves it in a temporary folder in ~/.vscode.

**Comment by doudou** [comment]
I'm going to merge this, and add the "nothing is silently done" as issue.

**Pull request merged**